kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/expressions/DefaultExpressionUtils.java

```diff
@@ -86,6 +101,74 @@ public boolean getBoolean(int rowId) {
         };
     }
 
+    static ColumnVector comparatorVector(
+            ColumnVector left,
+            ColumnVector right, VectorComparator vectorComparator) {
+        checkArgument(
+                left.getSize() == right.getSize(),
+                "Left and right operand have different vector sizes.");
+        return new ColumnVector() {
+            @Override
+            public DataType getDataType() { return BooleanType.BOOLEAN;}
+
+            @Override
+            public void close() {
+                Utils.closeCloseables(left, right);
+            }
+
+            @Override
+            public int getSize() { return left.getSize(); }
+            @Override
+            public boolean isNullAt(int rowId) {
+                return left.isNullAt(rowId) || right.isNullAt(rowId);
```

Comment:
In ticket https://github.com/delta-io/delta/issues/2541 , it is mentioned that we want to check against the following truth table

| Left | Right | Result |
| :---         |     :---:      |          ---: |
| null   | null  | True    |
| not null (A)     | null     | False     |
| null    | not null (B)     | False     |
| not null (A)     | not null (B)     | False     |
| not null (A)     | not null (A)     | True     |

But the current behavior is 

| Left | Right | Result |
| :---         |     :---:      |          ---: |
| null   | null  | False    |
| not null (A)     | null     | False     |
| null    | not null (B)     | False     |
| not null (A)     | not null (B)     | False     |
| not null (A)     | not null (A)     | True     |

Do we want to change the behavior? I am guessing we want to follow the truth table here https://spark.apache.org/docs/latest/sql-ref-null-semantics.html#:~:text=In%20order%20to%20compare%20the,both%20the%20operands%20are%20NULL%20. ?

Reply:
Follow the existing method (i.e the second table). The example I gave in #2541 is for the null safe equals.

Reply:
Thanks for the clarification, the existing code should follow the second table, once this PR is merged, I will add support for null safe equals.